apps/src/code-studio/pd/workshop_dashboard/attendance/session_attendance.jsx

```diff
@@ -2,39 +2,163 @@
  * Display and edit attendance for a workshop session, for display in a WorkshopAttendance tab.
  */
 import React from 'react';
+import $ from 'jquery';
+import _ from 'lodash';
 import SessionAttendanceRow from './session_attendance_row';
+import VisibilitySensor from '../components/visibility_sensor';
 import {Table} from 'react-bootstrap';
+import IdleTimer from 'react-idle-timer';
+
+// in milliseconds
+const REFRESH_DELAY = 5000;
+const IDLE_TIMEOUT = 60000;
+
+const styles = {
+  idle: {
+    opacity: .5
+  }
+};
 
 const SessionAttendance = React.createClass({
   propTypes: {
-    sessionId: React.PropTypes.number,
-    attendance: React.PropTypes.array.isRequired,
+    workshopId: React.PropTypes.number.isRequired,
+    sessionId: React.PropTypes.number.isRequired,
     adminOverride: React.PropTypes.bool,
-    onChange: React.PropTypes.func,
-    isReadOnly: React.PropTypes.bool
+    isReadOnly: React.PropTypes.bool,
+    onSaving: React.PropTypes.func.isRequired,
+    onSaved: React.PropTypes.func.isRequired
+  },
+
+  getInitialState() {
+    return {
+      loading: true,
+      attendance: undefined,
+      refreshInterval: undefined
+    };
+  },
+
+  componentDidMount() {
+    this.load();
+    this.startRefreshInterval();
+  },
+
+  componentWillUnmount() {
+    this.stopRefreshInterval();
+  },
+
+  startRefreshInterval() {
+    if (!this.state.refreshInterval) {
+      const refreshInterval = window.setTimeout(this.load, REFRESH_DELAY);
+      this.setState({refreshInterval});
```

Comment:
This is the ES6 [object matching shorthand](https://github.com/lukehoban/es6features#destructuring). It is equivalent to:
```javascript
this.setState({refreshInterval: refreshInterval});
```

Basically, when declaring an object, keys with no `:` pick up the value of the variable with that name from scope. So:

```ES6
const x = 5;
const y = 10;
const myObj = {x, y, z: 15};
console.log(myObj);
// logs "Object {x: 5, y: 10, z: 15}"
```


Similarly, the following will destructure the object into scope variables:
```ES6
const anotherObj = {a: 1, b: 2, c: 3};
const {a, b} = anotherObj; // assigns a and b in scope
console.log(`a: ${a}, b: ${b}`);
// Logs "a: 1, b: 2"
```